SlevomatCodingStandard/Sniffs/Classes/ClassStructureSniff.php

```diff
@@ -0,0 +1,478 @@
+		}
+
+		return $visibility ?? T_PUBLIC;
+	}
+
+	private function isMemberStatic(File $file, int $pointer): bool
+	{
+		$tokens = $file->getTokens();
+		$tokenTypes = [T_OPEN_CURLY_BRACKET, T_CLOSE_CURLY_BRACKET, T_STATIC];
+		$previousPointer = $file->findPrevious($tokenTypes, $pointer, null, false, null, true);
+
+		return $tokens[$previousPointer]['code'] === T_STATIC;
+	}
+
+	private function isStaticConstructor(File $file, int $pointer, bool $isStatic): bool
+	{
+		if (! $isStatic) {
+			return false;
+		}
+
+		if ($this->getVisibilityForToken($file, $pointer) !== T_PUBLIC) {
+			return false;
+		}
+
+		$parentClassName = $this->getParentClassName($file, $pointer);
+
+		$returnTypeHint = FunctionHelper::findReturnTypeHint($file, $pointer);
+		if ($returnTypeHint !== null) {
+			return in_array($returnTypeHint->getTypeHint(), ['self', $parentClassName], true);
+		}
+
+		$returnAnnotation = FunctionHelper::findReturnAnnotation($file, $pointer);
+		if ($returnAnnotation === null) {
+			return false;
+		}
+
+		return in_array($returnAnnotation->getContent(), ['static', 'self', $parentClassName], true);
+	}
+
+	private function getParentClassName(File $file, int $pointer): string
+	{
+		$classPointer = TokenHelper::findPrevious($file, Tokens::$ooScopeTokens, $pointer - 1);
+		if ($classPointer === null) {
+			return '';
+		}
+
+		return ClassHelper::getName($file, $classPointer);
+	}
+
+	private function fixInvalidStagePlacement(
+		File $file,
+		int $stageFirstMemberPointer,
+		int $stageLastMemberPointer,
+		int $nextStageMemberPointer
+	): void
+	{
+		$tokens = $file->getTokens();
+
+		$previousMemberEndPointer = $this->findPreviousMemberEndPointer($file, $stageFirstMemberPointer);
+
+		$stageStartPointer = $this->findStageStartPointer($file, $stageFirstMemberPointer, $previousMemberEndPointer);
+		$stageEndPointer = $this->findStageEndPointer($file, $stageLastMemberPointer);
+
+		$nextStageMemberStartPointer = $this->findStageStartPointer($file, $nextStageMemberPointer);
+
+		$linesBetween = $tokens[$stageStartPointer]['line'] - $tokens[$previousMemberEndPointer]['line'];
+
+		$file->fixer->beginChangeset();
+
+		$content = '';
+		for ($i = $stageStartPointer; $i <= $stageEndPointer; $i++) {
+			$content .= $tokens[$i]['content'];
+			$file->fixer->replaceToken($i, '');
+		}
+
+		$this->removeBlankLinesAfterMember($file, $linesBetween, $previousMemberEndPointer, $stageStartPointer);
+
+		$newLines = str_repeat($file->eolChar, $linesBetween);
+		$file->fixer->addContentBefore($nextStageMemberStartPointer, $content . $newLines);
+
+		$file->fixer->endChangeset();
+	}
+
+	private function findPreviousMemberEndPointer(File $file, int $memberPointer): int
+	{
+		$endTypes = [T_OPEN_CURLY_BRACKET, T_CLOSE_CURLY_BRACKET, T_SEMICOLON];
+		$previousMemberEndPointer = TokenHelper::findPrevious($file, $endTypes, $memberPointer - 1);
+		if ($previousMemberEndPointer === null) {
+			throw new RuntimeException('Previous member end pointer not found');
```

Comment:
Are we able to remove all `throw new RuntimeException` ?

Reply:
Sure, I'll replace it with asserts.